packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingUtils.ts

```diff
@@ -191,3 +195,41 @@ export const setStrategyAvailability = (
 
   apiRef.current.unstable_setStrategyAvailability('rowTree', ROW_GROUPING_STRATEGY, isAvailable);
 };
+
+export const getGroupingRules = ({
+  sanitizedRowGroupingModel,
+  columnsLookup,
+}: {
+  sanitizedRowGroupingModel: GridRowGroupingModel;
+  columnsLookup: GridColumnRawLookup;
+}): GridGroupingRules =>
+  sanitizedRowGroupingModel.map((field) => ({
+    field,
+    groupingValueGetter: columnsLookup[field]?.groupingValueGetter,
+  }));
+
+/**
+ * Compares two sets of grouping rules to determine if they are equal or not.
+ */
+export const hasGroupingRulesChanged = (
```

Comment:
Nitpick:
```suggestion
export const haveGroupingRulesChanged = (
```
?

Or inverse it to `areGroupingRulesEqual`?